File: problems/1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> hash;
        
        for (int i = 0; i < nums.size(); i++)
        {
            /* code */
            int t = target - nums[i];
            if (hash.count(t))
            {
                return vector<int>({hash[t],i});
            }
            hash[nums[i]] = i;
            
        }
        return vector<int>();
        
    }
};
// @lc code=end

File: problems/1.两数之和.java
import java.util.HashMap;
import java.util.Map;

/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hash = new HashMap<Integer,Integer>();
        for(int i = 0 ; i< nums.length; i++){
            if(hash.containsKey(target- nums[i])){
                return new int[]{hash.get(target-nums[i]), i};
            }
            hash.put(nums[i], i);
        }

        return new int[2];
    }
}
// @lc code=end

